cpp/oneapi/dal/backend/common.hpp

```diff
@@ -20,7 +20,7 @@
 #include <CL/sycl.hpp>
 #endif
 
-#include "oneapi/dal/backend/dispatcher.hpp"
+#include "oneapi/dal/detail/common.hpp"
```

Comment:
Does not that lead to a recursive dependency problem in the future between backend and details?

Reply:
`backend` cannot be included to `detail` by design, it will not even compile as backend is not distributed together with detail.